crates/bevy_core/src/lib.rs

```diff
@@ -46,6 +45,24 @@ impl Plugin for CorePlugin {
             .unwrap_or_else(DefaultTaskPoolOptions::default)
             .create_default_pools(app.world_mut());
 
+        #[cfg(not(target_arch = "wasm32"))]
+        app.add_system(
+            task_pool_options::handle_task_pool_panicking_threads_system::<bevy_tasks::IoTaskPool>
+                .system(),
+        )
+        .add_system(
+            task_pool_options::handle_task_pool_panicking_threads_system::<
+                bevy_tasks::ComputeTaskPool,
+            >
+                .system(),
+        )
+        .add_system(
+            task_pool_options::handle_task_pool_panicking_threads_system::<
+                bevy_tasks::AsyncComputeTaskPool,
+            >
+                .system(),
+        );
```

Comment:
I think this should be part of whichever plugin starts the task pool and not `CorePlugins`.

Reply:
core plugin is the one that starts the task pools.
Though arguable, we could refactor this out into its own plugin...

Reply:
that... is one of the ugliest formatting I've seen 😞

Reply:
yeppppp